src/vec32.rs

```diff
@@ -103,8 +104,20 @@ impl<T> Vec32<T> {
             self.len += 1;
         }
     }
+    #[inline]
+    pub fn extend_from_slice(&mut self, other: &[T]) {
+        if self.len + other.len() as u32 >= self.cap {
```

Comment:
A couple of edge cases to consider:

- If `other.len() > u32::MAX as usize` then `other.len() as u32` will silently truncate, and we will have insufficient capacity
- If `self.len + other.len()` would overflow, then we get a panic on debug, and insufficient capacity on release

Reply:
I updated the code to use `usize`, which should mitigate the overflow issue somewhat. It will still not work correctly when the passed slice is larger than 4GB